### ISSUE TITLE
Planner segfault when combining DISTINCT with EXECUTE ON ALL SEGMENTS

### ISSUE THREAD

**Issue by jchampio** [greenplum-db/gpdb#8357]
The crash is in `cost_common_agg()`, which assumes that `info->input_path->parent` is non-NULL. This is making it difficult to write a nice regression case for #8354...

### Greenplum version or build
master

#### Setup
```
postgres=# CREATE FUNCTION test()
    RETURNS SETOF boolean
    EXECUTE ON ALL SEGMENTS
    LANGUAGE plpgsql AS $$
BEGIN
    RETURN QUERY SELECT true;
END
$$;
CREATE FUNCTION

postgres=# SELECT test();
 test
------
 t
 t
 t
(3 rows)

postgres=# SELECT DISTINCT test();
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!>
```
#### Log
```
2019-08-12 16:13:28.267290 PDT,,,p7982,th0,,,2019-08-12 16:13:26 PDT,0,con5,cmd1,seg-1,,,,,"PANIC","XX000","Unexpected internal error: Master process received signal SIGSEGV",,,,,,,0,,,,"1    0x10b3bc8a4 postgres StandardHandlerForSigillSigsegvSigbus_OnMainThread + 0x204
2    0x10b209faf postgres CdbProgramErrorHandler + 0x13f
3    0x7fffb616ab3a libsystem_platform.dylib _sigtramp + 0x1a
4    0x10b155f80 postgres pull_var_clause_walker + 0x0
5    0x10b4837d8 postgres cost_1phase_aggregation + 0xb8
6    0x10b4821a7 postgres cdb_grouping_planner + 0xe77
7    0x10b116ff0 postgres grouping_planner + 0xd60
8    0x10b114c00 postgres subquery_planner + 0x950
9    0x10b113b66 postgres standard_planner + 0x436
10   0x10b113713 postgres planner + 0x163
11   0x10b2065f3 postgres pg_plan_query + 0x53
12   0x10b2066cd postgres pg_plan_queries + 0x6d
13   0x10b20a7a1 postgres exec_simple_query + 0x4e1
14   0x10b208f07 postgres PostgresMain + 0xe07
15   0x10b175b0d postgres BackendRun + 0x26d
16   0x10b174c18 postgres BackendStartup + 0x198
"
2019-08-12 16:13:28.267416 PDT,,,p81506,th-1092103232,,,,0,,,seg-1,,,,,"LOG","00000","server process (PID 7982) was terminated by signal 11: Segmentation fault","Failed process was running: select distinct test();",,,,,,0,,"postmaster.c",3961,
```

**Comment by jchampio** [comment]
This can be worked around with a subquery:
```
postgres=# select distinct q.* from (select test()) q;
 test
------
 t
(1 row)
```

**Comment by berlin-ab** [comment]
I can confirm that this is reproducible with @jchampio 's reproduction:

```
adamberlin=# select version();
LOG:  00000: statement: select version();
LOCATION:  exec_simple_query, postgres.c:1617
┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                 version                                                                                                                 │
├─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ PostgreSQL 9.4.20 (Greenplum Database 7.0.0-alpha.0+dev.704.gad8541b14db build dev) on x86_64-apple-darwin17.7.0, compiled by Apple LLVM version 8.1.0 (clang-802.0.42), 64-bit compiled on Aug 16 2019 09:13:24 (with assert checking) │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

```

**Comment by hlinnaka** [comment]
Richard fixed this on master by commit 7d79311c310281321f0ded0d48972354a6fe947f, and on 6X_STABLE by commit 3e2fcc08fb9a8703825d7bbe52f2f74e5e4c889a. Closing.
